get_CLN_data.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+chain=$1
+color_green=$2
+color_red=$3
+ln_git_version=$4
+
+
+echo -ne '\r### Loading CoreLN data \r'
+
+ln_running=$(systemctl is-active cln)
+coreln_color="${color_green}"
+if [ -z "${ln_running##*inactive*}" ]; then
+  ln_running="down"
+  coreln_color="${color_red}\e[7m"
+else
+  if [ -z "${ln_running##*failed*}" ]; then
+    ln_running="down"
+    coreln_color="${color_red}\e[7m"
+  else
+    ln_running="up"
+  fi
+fi
+lncli="/home/cln/lightning/cli/lightning-cli"
+
+printf "%0.s#" {1..63}
+echo -ne '\r### Loading CoreLN data \r'
+
+
+alias_color="${color_grey}"
+ln_alias="$(${lncli} getinfo | jq -r '.alias')" 2>/dev/null
+
+ln_walletbalance=0
+#check if len(outputs) == 0 --> empty wallet
+amount_of_wallet_transactions="$(${lncli} listfunds | jq -r '.outputs | length')"
+if [ $amount_of_wallet_transactions -gt 0 ];then
+ln_walletbalance=$(${lncli} listfunds | jq -r '.outputs[0].value | tonumber')
+fi
+
+ln_channelbalance="$(${lncli} listfunds | jq -r '.channels[0].our_amount_msat | gsub("msat";"")| tonumber' |  awk '{print $1/1000}' )" 2>/dev/null
+if [ -z "${ln_channelbalance}" ]; then
+  ln_channelbalance=0
+fi
+printf "%0.s#" {1..66}
+
+echo -ne '\r### Loading CoreLN data \r'
+
+ln_channels_online="$(${lncli} getinfo | jq -r '.num_active_channels')" 2>/dev/null
+ln_channels_total="$(${lncli} listincoming | jq '.[] | length')" 2>/dev/null
+node_id="$(${lncli} getinfo | jq -r '.id')" 2>/dev/null
+node_address="$(${lncli} getinfo | jq -r '.address[0].address')" 2>/dev/null
+node_port="$(${lncli} getinfo | jq -r '.address[0].port')" 2>/dev/null
+ln_connect_addr="$node_id"@"$node_address":"$node_port" 2>/dev/null
+ln_connect_guidance="lightning-cli connect ${ln_connect_addr}"
+ln_external="$(echo "${ln_connect_addr}" | tr "@" " " |  awk '{ print $2 }')" 2>/dev/null
+if [ -z "${ln_external##*onion*}" ]; then
+  ln_external="Using TOR Address"
+fi
+
+printf "%0.s#" {1..70}
+# echo -ne '\r### Loading LND data \r'
+
+# ln_pendingopen=$($lncli pendingchannels  | jq '.pending_open_channels[].channel.local_balance|tonumber ' | awk '{sum+=$0} END{print sum}')
+# if [ -z "${ln_pendingopen}" ]; then
+#   ln_pendingopen=0
+# fi
+
+# ln_pendingforce=$($lncli pendingchannels  | jq '.pending_force_closing_channels[].channel.local_balance|tonumber ' | awk '{sum+=$0} END{print sum}')
+# if [ -z "${ln_pendingforce}" ]; then
+#   ln_pendingforce=0
+# fi
+
+# ln_waitingclose=$($lncli pendingchannels  | jq '.waiting_close_channels[].channel.local_balance|tonumber ' | awk '{sum+=$0} END{print sum}')
+# if [ -z "${ln_waitingclose}" ]; then
+#   ln_waitingclose=0
+# fi
+
+# echo -ne '\r### Loading LND data \r'
+
+# ln_pendinglocal=$((ln_pendingopen + ln_pendingforce + ln_waitingclose))
+ln_pendinglocal=0
```

Comment:
when you type `lightning-cli listfunds` and check the chanel object it will look simiar to this:
```
"channels": [
      {
         "peer_id": "XXX",
         "connected": true,
         "state": "CHANNELD_NORMAL",
         "short_channel_id": "YYY",
         "channel_sat": 9999,
         "our_amount_msat": "9999999msat",
         "channel_total_sat": 9999,
         "amount_msat": "9999999msatmsat",
         "funding_txid": "XXX",
         "funding_output": 1
      }
   ]
   ```
   On could sum up all chanels where `state  !=  "CHANNELD_NORMAL"`, i was just to lazy to dig more into this